velox/common/encode/Base64.h

```diff
@@ -89,19 +88,24 @@ class Base64 {
   static size_t
   decode(const char* src, size_t src_len, char* dst, size_t dst_len);
 
-  static void decodeUrl(
-      const char* src,
-      size_t src_len,
-      char* dst,
-      size_t dst_len,
-      bool pad);
+  static void
+  decodeUrl(const char* src, size_t src_len, char* dst, size_t dst_len);
 
   constexpr static char kBase64Pad = '=';
 
  private:
+  static inline bool isPadded(const char* data, size_t len) {
+    return (len > 0 && data[len - 1] == kBase64Pad) ? true : false;
+  }
+
   static inline size_t countPadding(const char* src, size_t len) {
-    DCHECK_GE(len, 2);
-    return src[len - 1] != kBase64Pad ? 0 : src[len - 2] != kBase64Pad ? 1 : 2;
+    size_t numPadding{0};
+    while (len > 0 && src[len - 1] == kBase64Pad) {
```

Comment:
This code seems to allow more than 2 padding characters, but the format allows only 1 or 2.

Reply:
The plan was to make `countPadding` and `isPadded` common for base32 and base64. We will make a common decision on updating both the function.